contracts/examples/Venus.sol

```diff
@@ -18,69 +18,144 @@
 pragma solidity ^0.8.9;
 
 import {ICrossL2Prover} from "../interfaces/ICrossL2Prover.sol";
+import {RLPReader} from "optimism/libraries/rlp/RLPReader.sol";
+import {Bytes} from "optimism/libraries/Bytes.sol";
 
 /**
  * @title Venus
  * @notice Venus is a simple polymer proof api contract that proves an event happened on another chain.
  * @dev This contract is used for only testing and as an example for dapp developers on how to
- * integrate with polymer's proof api.
+ * integrate with polymer's proof api. And some necessary validation which needs to occur
  */
 contract Venus {
     ICrossL2Prover public immutable prover;
+    address public immutable counterParty; // The dapp on the counter party chain we wish to prove on this local chain
+    bytes32 public lastReceivedTransmission; // Last received arguments from transmitted event
+    bytes32 public immutable chainId;
 
-    event SuccessfulReceipt(bytes receiptIndex, bytes receiptRLP);
-    event SuccessfulEvent(uint256 eventIndex, address sender);
+    // The event that we emit on source chain to be proven on this local chain
+    event TransmitToHouston(bytes32 message, uint64 timestamp);
+
+    // Event that we emit on this local chain to indicate that we have received an event from the source chain
+    event TransmissionReceived(bytes32 message, uint64 timestamp);
+
+    event SuccessfulReceipt(bytes32 srcChainId, bytes receiptRLP);
+    event ValidCounterpartyEvent(address counterParty, bytes[] topics, bytes unindexed);
 
     error invalidProverAddress();
     error invalidReceiptProof();
+    error invalidChainId(); // The chain id of the proof is not the expected chain id. It might be a valid event, but is
+        // on a different chain than we were expecting.
     error invalidEventProof();
+    error invalidEventSender();
+    error invalidCounterpartyEvent();
 
-    constructor(ICrossL2Prover _prover) {
+    constructor(ICrossL2Prover _prover, address _counterParty, bytes32 _chainId) {
         if (address(_prover) == address(0)) {
             revert invalidProverAddress();
         }
         prover = _prover;
+        counterParty = _counterParty;
```

Comment:
would it be easier/more flexible to pass these as parameters to `receiveTransmissionEvent`